packages/nx/src/native/index.js

```diff
@@ -1,268 +1,355 @@
-const { existsSync, readFileSync } = require('fs')
-const { join } = require('path')
+// prettier-ignore
+/* eslint-disable */
+/* auto-generated by NAPI-RS */
 
-const { platform, arch } = process
+const { readFileSync } = require('fs')
 
 let nativeBinding = null
-let localFileExisted = false
-let loadError = null
+const loadErrors = []
 
-function isMusl() {
-  // For Node 10
-  if (!process.report || typeof process.report.getReport !== 'function') {
-    try {
-      const lddPath = require('child_process').execSync('which ldd').toString().trim();
-      return readFileSync(lddPath, 'utf8').includes('musl')
-    } catch (e) {
+const isMusl = () => {
+  let musl = false
+  if (process.platform === 'linux') {
+    musl = isMuslFromFilesystem()
+    if (musl === null) {
+      musl = isMuslFromReport()
+    }
+    if (musl === null) {
+      musl = isMuslFromChildProcess()
+    }
+  }
+  return musl
+}
+
+const isFileMusl = (f) => f.includes('libc.musl-') || f.includes('ld-musl-')
+
+const isMuslFromFilesystem = () => {
+  try {
+    return readFileSync('/usr/bin/ldd', 'utf-8').includes('musl')
+  } catch {
+    return null
+  }
+}
+
+const isMuslFromReport = () => {
+  const report = typeof process.report.getReport === 'function' ? process.report.getReport() : null
+  if (!report) {
+    return null
+  }
+  if (report.header && report.header.glibcVersionRuntime) {
+    return false
+  }
+  if (Array.isArray(report.sharedObjects)) {
+    if (report.sharedObjects.some(isFileMusl)) {
       return true
     }
-  } else {
-    const { glibcVersionRuntime } = process.report.getReport().header
-    return !glibcVersionRuntime
   }
+  return false
 }
 
-switch (platform) {
-  case 'android':
-    switch (arch) {
-      case 'arm64':
-        localFileExisted = existsSync(join(__dirname, 'nx.android-arm64.node'))
-        try {
-          if (localFileExisted) {
-            nativeBinding = require('./nx.android-arm64.node')
-          } else {
-            nativeBinding = require('@nx/nx-android-arm64')
-          }
-        } catch (e) {
-          loadError = e
-        }
-        break
-      case 'arm':
-        localFileExisted = existsSync(join(__dirname, 'nx.android-arm-eabi.node'))
-        try {
-          if (localFileExisted) {
-            nativeBinding = require('./nx.android-arm-eabi.node')
-          } else {
-            nativeBinding = require('@nx/nx-android-arm-eabi')
```

Comment:
These imports have been broken some how.

The prefix is specified in `package.json` here: https://github.com/nrwl/nx/blob/fc8d5ba828118300bf4c4927cdb7578c46b5b355/packages/nx/package.json#L170 but is not being picked up by napi for some reason. Could you take a look please? :eyes: I imagine it's a result of upgrading Napi to the 3.0.0 version.

Reply:
@Brooooooklyn Do you know what might be going on here?

Reply:
I'll take a look tomorrow 🥹

Reply:
@FrozenPandaz it should be `napi.packageName` now